ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx

```diff
@@ -1,16 +1,131 @@
 
 import PageTitle from 'Components/PageTitle';
 import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
-import { complianceEnhancedCoveragePath } from 'routePaths';
+import TableErrorComponent from 'Components/PatternFly/TableErrorComponent';
+import NotFoundMessage from 'Components/NotFoundMessage';
+import DetailsPageHeader, { PageHeaderLabel } from './components/DetailsPageHeader';
+
+const STATUS_LABELS: Record<ComplianceCheckStatus, string> = {
+    FAIL: 'Fail',
+    INFO: 'Info',
+    PASS: 'Pass',
+    ERROR: 'Error',
+    MANUAL: 'Manual',
+    INCONSISTENT: 'Inconsistent',
+    NOT_APPLICABLE: 'Not applicable',
+    UNSET_CHECK_STATUS: 'Unset check status',
+};
+
+function getCheckStatIcon(status: ComplianceCheckStatus) {
+    switch (status) {
+        case 'FAIL':
+            return <SecurityIcon color="red" />;
+        case 'INFO':
+            return <InfoIcon />;
+        case 'PASS':
+            return <CheckCircleIcon color="blue" />;
+        case 'ERROR':
+            return <ExclamationTriangleIcon />;
+        case 'MANUAL':
+            return <WrenchIcon />;
+        case 'INCONSISTENT':
+            return <UnknownIcon />;
+        case 'NOT_APPLICABLE':
+            return <BanIcon />;
+        default:
+            return null;
+    }
+}
+
+function getCheckStatColor(status: ComplianceCheckStatus) {
+    switch (status) {
+        case 'FAIL':
+            return 'red';
+        case 'PASS':
+            return 'blue';
+        default:
+            return 'grey';
+    }
+}
+
+function sortCheckStats(a: ComplianceCheckStatusCount, b: ComplianceCheckStatusCount) {
+    const order: ComplianceCheckStatus[] = [
+        'PASS',
+        'FAIL',
+        'MANUAL',
+        'ERROR',
+        'INFO',
+        'NOT_APPLICABLE',
+        'INCONSISTENT',
+    ];
+    return order.indexOf(a.status) - order.indexOf(b.status);
+}
 
 function CheckDetails() {
     const { checkName, profileName } = useParams();
 
     const complianceCoverageChecksURL = `${complianceEnhancedCoveragePath}/profiles/${profileName}/checks`;
 
+    const profileCheckByIdFn = useCallback(
+        () => getComplianceProfileCheckStats(profileName, checkName),
+        [profileName, checkName]
+    );
+    const { data, loading: isLoading, error } = useRestQuery(profileCheckByIdFn);
+
+    if (error) {
+        return (
+            <PageSection variant="light">
+                <TableErrorComponent
+                    error={error}
+                    message="An error occurred. Try refreshing again"
+                />
+            </PageSection>
+        );
+    }
+
+    if (!isLoading && !data) {
+        return (
+            <NotFoundMessage
+                title="404: Profile check does not exist."
+                message={`A profile check called "${checkName}" could not be found.`}
+            />
+        );
+    }
+
+    const checkStatsLabels =
```

Comment:
I think most of the code related to the labels can be simplified using the `getClusterResultsStatusObject` in `compliance.coverage.utils.tsx`. I think it makes sense to add `labelColor` to the object then all you really have to do is something like this to get the same effect:

```javascript
<LabelGroup numLabels={7}>
    {data?.checkStats.map((checkStat) => {
        const statusObject = getClusterResultsStatusObject(checkStat.status);
        return (
            <Label
                variant="filled"
                icon={statusObject.icon}
                color={statusObject.labelColor}
            >
                {statusObject.statusText}: {checkStat.count}
            </Label>
        );
    })}
</LabelGroup>
```

Reply:
Sweet, that works for me 👍🏼